src/main/java/tech/toparvion/analog/util/config/ChoicesCustomConfigurationLoader.java

```diff
@@ -0,0 +1,80 @@
+package tech.toparvion.analog.util.config;
+
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.boot.env.PropertySourceLoader;
+import org.springframework.boot.env.YamlPropertySourceLoader;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertySource;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.springframework.util.CollectionUtils.isEmpty;
+
+/**
+ * Environment post processor capable of loading choices list from a custom file.<br/>
+ *
+ * @author Polyudov
+ * @implNote <ul>
+ * <li>There are no logs because the logging subsystem is not initialized yet</li>
+ * <li>Works only with <code>.yaml/.yml</code> files</li>
+ * </ul>
+ * @since v0.14
+ */
+public class ChoicesCustomConfigurationLoader implements EnvironmentPostProcessor {
+  private static final String CHOICES_RESOURCE_NAME = "choicesCustomResource";
+
+  private final PropertySourceLoader loader;
+
+  @SuppressWarnings("unused") //for Spring
+  public ChoicesCustomConfigurationLoader() {
+    this.loader = new YamlPropertySourceLoader();
+  }
+
+  @Override
+  public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+    String choicesPath = environment.getProperty("choices-source.location");
+    if (isNullOrEmpty(choicesPath)) {
+      return;
+    }
+
+    FileSystemResource resource = new FileSystemResource(choicesPath);
+    if (!resource.isReadable() || !isExtensionCorrect(resource.getFilename())) {
+      return;
```

Comment:
This is quite important execution way to emit a log message upon reaching (in spite of log system absence). I think even a trivial `System.err.println("..."):` will be enough to significantly ease the investigation of what's going wrong.

Reply:
Done